openplacereviews-core/src/main/java/org/openplacereviews/osm/service/MergePlaceBot.java

```diff
@@ -211,6 +213,141 @@ protected void mergePlaces(OprMapCollectionApiResult res, MergeInfo info) {
+				if (OsmMapUtils.getDistance(latLonDeleted.getLatitude(), latLonDeleted.getLongitude(), featurePoint.lat(), featurePoint.lon()) < 150) {
+					res.add(feature);
+				}
+			}
+		}
+		return res;
+	}
+
+	private List<Feature> filterByOsmId(List<Feature> placesToMerge, OpObject deleted) {
+		String osmIdDeleted = getOsmId(deleted);
+		List<Feature> res = new ArrayList<>();
+		if (osmIdDeleted != null) {
+			for (Feature feature : placesToMerge) {
+				if (!feature.properties().get(OSM_ID).toString().equals(osmIdDeleted)) {
+					res.add(feature);
+				}
+			}
+		}
+		return res;
+	}
+
+	private boolean foundPlaceWithSameName(List<Feature> placesToMerge, OpObject deleted) {
+		List<Map<String, String>> placesToMergeTags = new ArrayList<>();
+		Map<String, String> oldOsmTags = getMainOsmTags(deleted);
+		if (oldOsmTags != null) {
+			List<OpObject> objToMerge = new ArrayList<>();
+			for (Feature feature : placesToMerge) {
+				objToMerge.add(getCurrentObject(feature));
+			}
+			for (int i = 0; i < placesToMerge.size(); i++) {
+				placesToMergeTags.add(getMainOsmTags(objToMerge.get(i)));
+			}
+			for (MatchType mt : EnumSet.allOf(MatchType.class)) {
+				int matched = -1;
+				for (int i = 0; i < placesToMerge.size(); i++) {
+					if (match(mt, oldOsmTags, placesToMergeTags.get(i))) {
+						if (matched >= 0) {
+							if (mt.allow2PlacesMerge) {
+								matched = i;
+							} else {
+								return false;
+							}
+						} else {
+							matched = i;
+						}
+					}
+				}
+				if (matched >= 0) {
+					return true;
+				}
+			}
+		}
+		return false;
+	}
+
+	private List<Feature> getNoDeletedPlaces(List<Feature> features) {
+		List<Feature> res = new ArrayList<>();
+		for (Feature feature : features) {
+			if (!feature.properties().containsKey(F_DELETED_PLACE)) {
+				Map<String, Object> osm = getMainOsmFromList(getCurrentObject(feature));
+				if (osm != null && !osm.containsKey(F_DELETED_OSM)) {
+					res.add(feature);
+				}
+			}
+		}
+		return res;
+	}
+
+	private LocalDate transformStringDateToLocaleDate(String date) {
+		return LocalDate.parse(date.substring(0, date.indexOf("+")), DateTimeFormatter.ISO_LOCAL_DATE_TIME);
+	}
+
+	private LocalDate getDeletedDate(OpObject deleted) {
+		Map<String, Object> mainOsm = getMainOsmFromList(deleted);
+		if (mainOsm != null) {
+			for (Map.Entry<String, Object> entry : mainOsm.entrySet()) {
+				if (entry.getKey().equals(F_DELETED_OSM)) {
+					return transformStringDateToLocaleDate((String) entry.getValue());
+				}
+			}
+		}
+		return null;
+	}
+
+	private boolean wasDeletedMoreThanTenDaysAgo(OpObject deleted) {
+		LocalDate dateToday = LocalDate.now();
+		LocalDate dateDeleted = getDeletedDate(deleted);
+
+		if (dateDeleted != null) {
+			return Duration.between(dateDeleted.atStartOfDay(), dateToday.atStartOfDay()).toDays() >= 10;
+		} else {
+			return false;
+		}
+
+	}
+
+	private OprMapCollectionApiResult getDataReport(OpObject deleted) {
```

Comment:
it is better & nicer method private OprMapCollectionApiResult getDataReport(String tileId) {